src/library/trackcollectionmanager.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include <QDir>
+#include <QList>
+#include <QObject>
+#include <QSet>
+
+#include "library/scanner/libraryscanner.h"
+#include "preferences/usersettings.h"
+#include "track/globaltrackcache.h"
+#include "util/db/dbconnectionpool.h"
+
+class TrackCollection;
+class ExternalTrackCollection;
+
+// Manages the internal and a varying number of external track collections.
+//
+// All modifying operations that might affect external collections
+// must be invoked through this class to keep all track collections
+// synchronized!
+//
+// Both crates and playlists are currently only supported by the internal
+// collection, which needs to be modified directly.
+class TrackCollectionManager: public QObject,
+    public virtual /*implements*/ GlobalTrackCacheSaver {
+    Q_OBJECT
+
+  public:
+    TrackCollectionManager(
+            QObject* parent,
+            UserSettingsPointer pConfig,
+            mixxx::DbConnectionPoolPtr pDbConnectionPool,
+            deleteTrackFn_t deleteTrackFn = nullptr);
+    ~TrackCollectionManager() override;
+
+    TrackCollection* internalCollection() {
```

Comment:
Add a comment about this function being deprecated here?

Reply:
Can't be deprecated, too early. Read-only functions are still accessed directly and we need the pointer for signal/slot connections.

Reply:
Is it a long term goal to remove this function? If so, it may still be helpful to document that so new code avoids it as much as practical.